app/Site/FortifyServiceProvider.php

```diff
@@ -45,6 +52,38 @@ public function register(): void
         Fortify::ignoreRoutes();
 
         $this->app->singleton(LoginResponseContract::class, LoginResponse::class);
+
+        Fortify::authenticateThrough(function (Request $request) {
+            return array_filter([
+                config('fortify.limiters.login') ? null : EnsureLoginIsNotThrottled::class,
+                Features::enabled(Features::twoFactorAuthentication()) ? RedirectIfTwoFactorAuthenticatable::class : null,
+                function ($request, $next) {
+                    $user = User::firstWhere(Fortify::username(), $request->input(Fortify::username()));
+
+                    // banned users should not have a password anymore. make sure they cannot get back in when a password still exists
+                    if ($user && $user->getAttribute('Permissions') < Permissions::Unregistered) {
+                        throw ValidationException::withMessages([
+                            Fortify::username() => [trans('auth.failed')],
+                        ]);
+                    }
```

Comment:
This is not necessary as long as passwords are properly cleared when role changes are applied. I'll look into that separately.